sycl/include/sycl/handler.hpp

```diff
@@ -1028,6 +1054,21 @@ class __SYCL_EXPORT handler {
                                 KernelType KernelFunc) {
     throwIfActionIsCreated();
     throwOnLocalAccessorMisuse<KernelName, KernelType>();
+
+    // Aside from the OpenCL CPU backend, the other backends cannot
+    // handle kernel launches when one or more of the dimensions of
+    // the global range exceeds the 32-bit max (*). However, in sycl,
+    // the global range size is specified by a size_t, which can be up
+    // to 64 bits. So, when we get such a global range, we wrap the
+    // old kernel in a new kernel that has each work item peform
+    // multiple invocations the old kernel in a 32-bit global range.
+#ifdef __SYCL_ENABLE_LARGE_RANGE_WRAPPING__
+    for (int i = 0; i < Dims; ++i)
+      if (NumWorkItems[i] > (std::numeric_limits<uint32_t>::max)())
```

Comment:
Unrelated to the PR, but I wonder if we want to have support for `any_of` over `range`. Because at first I was confused about why do we launch `parallel_for` for each dimension until I noticed `return`.

Having these lines in form of 
```c++
if (any_of(NumWorkItems, [](size_t item) {
    return item > (std::numeric_limits<uint32_t>::max)(); }))
  return parallel_for_large_range...
```

would improve readability, I believe.

Once again, this is a topic for a separate refactoring/improvement within the runtime and only if we decide to go this way.